pkg/engine/lifecycletest/framework/framework.go

```diff
@@ -59,6 +57,26 @@ import (
 	"github.com/pulumi/pulumi/sdk/v3/go/common/workspace"
 )
 
+// TB is a subset of testing.TB that admits other T-like things, such as *rapid.T from the Rapid property-testing
+// library. It covers the set of functionality that we actually need for lifecycle testing, and satisfies the interfaces
+// of testify.assert and testify.require.
+type TB interface {
```

Comment:
So I know this is the name in the framework but I want to just ask:

Are we certain this (terrible but consistent) name is worth keeping? Perhaps we should consider something more descriptive like TestingSharedFunctionality (equally bad).

They're both not great but figured I'd raise it

Reply:
Discussed offline and I think we're going to keep this as-is since `TB` is something of a standard in the Go language itself.